api/v1alpha1/contour_types.go

```diff
@@ -288,6 +288,21 @@ type ProviderLoadBalancerParameters struct {
 	// +unionDiscriminator
 	// +kubebuilder:default=AWS
 	Type LoadBalancerProviderType `json:"type,omitempty"`
+
+	// AWS specific LoadBalancer parameters.
```

Comment:
I suggest adding "Present only if type is AWS" so users understand that this field has a dependency. Repeat for other providers.

Reply:
Fixed for all providers.